rusoto/credential/src/environment.rs

```diff
@@ -104,31 +104,31 @@ impl EnvironmentProvider {
 }
 
 /// A private trait for building the environment variable names based
-/// on a provided prefix. Pulled out of the base `EnvironmentProvider`
-/// to keep separate from the public functionality.
+/// on a provided prefix. Smallest subset of functionality needed for
+/// Credentials building (see `credentials_from_environment` below).
 trait EnvironmentVariableProvider {
-    fn prefix(self: &Self) -> String;
+    fn prefix(&self) -> &str;
 
-    fn access_key_id_var(self: &Self) -> String {
+    fn access_key_id_var(&self) -> String {
         format!("{}_ACCESS_KEY_ID", self.prefix())
     }
 
-    fn secret_access_key_var(self: &Self) -> String {
+    fn secret_access_key_var(&self) -> String {
         format!("{}_SECRET_ACCESS_KEY", self.prefix())
     }
 
-    fn session_token_var(self: &Self) -> String {
+    fn session_token_var(&self) -> String {
         format!("{}_SESSION_TOKEN", self.prefix())
     }
 
-    fn credential_expiration_var(self: &Self) -> String {
+    fn credential_expiration_var(&self) -> String {
         format!("{}_CREDENTIAL_EXPIRATION", self.prefix())
     }
 }
 
 impl EnvironmentVariableProvider for EnvironmentProvider {
-    fn prefix(self: &Self) -> String {
-        self.prefix.clone()
+    fn prefix(&self) -> &str {
+        self.prefix.as_str()
```

Comment:
Sorry for the delay. To use a `&str` here I'm using this coercion here. I'm not sure if `as_str()` clones the `String` data or not. The alternative is to force `EnvironmentProvider` to have a lifetime scoped to the internal string slice. But that would have been a much more involved change that would effect the `ChainProvider` etc. If that's the way you want me to go, I can totally update it, just wanted to get you an update as soon as possible.

Reply:
No problem! Using `as_str` here is fine and doesn't clone the data (using `&self.prefix` would have worked as well).